src/module/item/consumable/spell-consumables.ts

```diff
@@ -81,8 +81,20 @@ async function createConsumableFromSpell(
     consumableSource.system.traits.value.push(...spell.traditions);
     consumableSource.name = getNameForSpellConsumable(type, spell.name, heightenedLevel);
     const description = consumableSource.system.description.value;
-    consumableSource.system.description.value =
-        (spell.sourceId ? "@" + spell.sourceId.replace(".", "[") + "]" : spell.description) + `\n<hr />${description}`;
+
+    consumableSource.system.description.value = (() => {
+        const paragraphElement = document.createElement("p");
+        const linkElement = document.createElement("em");
+        linkElement.append(spell.sourceId ? "@" + spell.sourceId.replace(".", "[") + "]" : spell.description);
+        paragraphElement.append(linkElement);
+
+        const containerElement = document.createElement("div");
+        const hrElement = document.createElement("hr");
+        containerElement.append(paragraphElement, hrElement);
+        hrElement.insertAdjacentHTML("afterend", description);
```

Comment:
This looks like it would nest a p element inside a p element, which can't be done. Could you paste the generated HTML?

Reply:
```
<p><em>@Compendium[pf2e.spells-srd.gKKqvLohtrSJj3BM]</em></p><hr><p>This baton is about a foot long and contains a single spell. The appearance typically relates to the spell within.</p>
<p><strong>Activate</strong> Cast a Spell</p>
<p><strong>Frequency</strong> once per day, plus overcharge</p>
<hr>
<p><strong>Effect</strong> You Cast the Spell at the indicated level.</p>
<p><strong>Craft Requirements</strong> Supply a listed-level casting of the spell.</p>
<hr>
<p><em>Note: To create a scroll or wand of a specific spell, drag the spell from the compendium or compendium browser into the inventory of a PC, NPC, or loot actor.</em></p>
```

```
<p><em>@Compendium[pf2e.spells-srd.rfZpqmj0AIIdkVIs]</em></p><hr><p>Casting a Spell from a scroll requires holding the scroll in one hand and activating it with a Cast a Spell activity using the normal number of actions for that spell.</p>
<p>To Cast a Spell from a scroll, the spell must appear on your spell list. Because you're the one Casting the Spell, use your spell attack roll and spell DC. The spell also gains the appropriate trait for your tradition (arcane, divine, occult, or primal).</p>
<p>Any physical material components and costs are provided when a scroll is created, so you don't need to provide them when Casting a Spell from a scroll. You must replace any required material component for that spell with a somatic component. If the spell requires a focus, you must have that focus to Cast the Spell from a scroll.</p>
<hr>
<p><em>Note: To create a scroll or wand of a specific spell, drag the spell from the compendium or compendium browser into the inventory of a PC, NPC, or loot actor.</em></p>
```